rinad/src/ipcm/app-handlers.cc

```diff
@@ -47,61 +47,68 @@ void IPCManager_::os_process_finalized_handler(
 			<< "terminated" << endl;
 	FLUSH_LOG(INFO, ss);
 
-	//Prevent any insertion/deletion to happen
-	rina::ReadScopedLock readlock(ipcp_factory_.rwlock);
-
-	// Look if the terminating application has allocated flows
-	// with some IPC processes
-	collect_flows_by_application(app_name, involved_flows);
-	unsigned short ipcp_id = 0;
-	for (list<rina::FlowInformation>::iterator fit = involved_flows.begin();
-			fit != involved_flows.end(); fit++) {
-
-		IPCMIPCProcess *ipcp = select_ipcp_by_dif(fit->difName);
-		if (!ipcp) {
-			ss  << ": Cannot find the IPC process "
-					"that provides the flow with port-id " <<
-					fit->portId << endl;
-			FLUSH_LOG(ERR, ss);
-			continue;
-		}
+	{
+		//Prevent any insertion/deletion to happen
+		rina::ReadScopedLock readlock(ipcp_factory_.rwlock);
+
+		// Look if the terminating application has allocated flows
+		// with some IPC processes
+		collect_flows_by_application(app_name, involved_flows);
+		unsigned short ipcp_id = 0;
+		for (list<rina::FlowInformation>::iterator fit = involved_flows.begin();
+				fit != involved_flows.end(); fit++) {
+
+			IPCMIPCProcess *ipcp = select_ipcp_by_dif(fit->difName);
+			if (!ipcp) {
+				ss  << ": Cannot find the IPC process "
+						"that provides the flow with port-id " <<
+						fit->portId << endl;
+				FLUSH_LOG(ERR, ss);
+				continue;
+			}
 
-		{
-			//Auto release the read lock
-			rina::ReadScopedLock readlock(ipcp->rwlock, false);
-			ipcp_id = ipcp->get_id();
-		}
+			{
+				//Auto release the read lock
+				rina::ReadScopedLock readlock(ipcp->rwlock, false);
+				ipcp_id = ipcp->get_id();
```

Comment:
Vincenzo, this is pull request is about a minor fix to improve the stability of the current IPCM. The former pull request (aka the monster) was using this design extensively (in almost all the functions). The code in this pull request just follows the same design that is already there. 

We can discuss if it needs to improved and be made less obscure, but this would be another IPCM refactor. Therefore I would just merge this humble pull request in, which will make the live of people using the stack a bit easier ;)